source/server/listener_manager_impl.cc

```diff
@@ -305,6 +306,20 @@ ListenerManagerStats ListenerManagerImpl::generateStats(Stats::Scope& scope) {
 
 bool ListenerManagerImpl::addOrUpdateListener(const envoy::api::v2::Listener& config,
                                               const std::string& version_info, bool added_via_api) {
+
+  // TODO(junr03): currently only one ApiListener can be installed via bootstrap to avoid having to
+  // build a collection of listeners, and to have to be able to warm and drain the listeners. In the
+  // future allow multiple ApiListeners, and allow them to be created via LDS as well as bootstrap.
+  if (!api_listener_ && !added_via_api && config.has_api_listener()) {
```

Comment:
need to figure out why the listener manager unit tests generate proto structs that _have_ an api listener. Based on my dive I can't see how to control default creation with the protobuf utils.